service-config/src/main/java/com/palantir/config/service/ServiceDiscoveryConfiguration.java

```
@@ -0,0 +1,121 @@
+     * @param serviceName
+     *        the name of the service to be used to retrieve the respective {@link ServiceConfiguration}
+     * @returns the API token for a service based on the input service name
+     */
+    public final Optional<BearerToken> getApiToken(String serviceName) {
+        checkArgument(!Strings.isNullOrEmpty(serviceName));
+
+        Optional<ServiceConfiguration> maybeServiceConfig = getServiceConfiguration(serviceName);
+        if (!maybeServiceConfig.isPresent()) {
+            return defaultApiToken();
+        }
+
+        return maybeServiceConfig.get().apiToken().or(defaultApiToken());
+    }
+
+    /**
+     * Uses the service name as the key to retrieve the {@link SSLConfiguration} for the respective
+     * {@link ServiceConfiguration}. If the {@link SslConfiguration} is not defined in the {@link ServiceConfiguration},
+     * the default {@link SslConfiguration} will be returned.
+     *
+     * @param serviceName
+     *        the name of the service to be used to retrieve the respective {@link ServiceConfiguration}
+     * @returns the {@link ServiceConfiguration} for a service based on the input service name
+     */
+    public final Optional<SslConfiguration> getSslConfiguration(String serviceName) {
+        checkArgument(!Strings.isNullOrEmpty(serviceName));
+
+        Optional<ServiceConfiguration> maybeServiceConfig = getServiceConfiguration(serviceName);
+        if (!maybeServiceConfig.isPresent()) {
+            return defaultSecurity();
+        }
+
+        return maybeServiceConfig.get().security().or(defaultSecurity());
+    }
+
+    /**
+     * Uses the service name as the key to retrieve the respective {@link ServiceConfiguration}.
+     *
+     * @param serviceName
+     *        the name of the service.
+     * @returns the {@link ServiceConfiguration} for a service based on the input service name
+     */
+    public final Optional<ServiceConfiguration> getServiceConfiguration(String serviceName) {
```

probably easier to return the nullable thing here, and let it throw RuntimeException when you try to use it later?